packages/core/src/App/Containers/NotificationsDialog/notifications-dialog.jsx

```diff
@@ -41,62 +41,67 @@ class NotificationsDialog extends React.Component {
+                                                [`notifications-item__title-icon--${item.type}`]: item.type,
+                                            })}
+                                        />
+                                    )}
+                                    {item.header}
+                                </h2>
+                                <div className='notifications-item__message'>{item.message}</div>
+                                <div className='notifications-item__action'>
+                                    {!isEmptyObject(item.action) && (
+                                        <>
+                                            {item.action.route ? (
+                                                <BinaryLink
+                                                    onClick={this.props.toggleDialog}
+                                                    active_class='notifications-item'
+                                                    className={classNames(
+                                                        'dc-btn',
+                                                        'dc-btn--secondary',
+                                                        'notifications-item__cta-button'
+                                                    )}
+                                                    to={item.action.route}
+                                                >
+                                                    <span className='dc-btn__text'>{item.action.text}</span>
+                                                </BinaryLink>
+                                            ) : (
+                                                <Button
+                                                    className={classNames(
+                                                        'dc-btn--secondary',
+                                                        'notifications-item__cta-button'
+                                                    )}
+                                                    onClick={item.action.onClick}
+                                                    text={item.action.text}
+                                                />
+                                            )}
+                                        </>
+                                    )}
+                                </div>
                             </div>
-                        </div>
-                    ))
+                        ))
                 ) : (
                     <EmptyNotification />
                 )}
             </React.Fragment>
         );
 
-        const is_empty = !(this.props.notifications && this.props.notifications.length);
+        const is_empty = !(
+            this.props.notifications.filter(message => message.type !== 'dp2p') &&
+            this.props.notifications.filter(message => message.type !== 'dp2p').length
```

Comment:
we're doing the same filtering many times, can we do it once and use it in these three places?

Reply:
Will do.